profiler/src/ProfilerEngine/Datadog.Profiler.Native.Linux/CrashReportingLinux.cpp

```diff
@@ -168,68 +194,96 @@ std::vector<StackFrame> CrashReportingLinux::GetThreadFrames(int32_t tid, Resolv
+                auto demangleResult = ddog_demangle(libdatadog::FfiHelper::StringToCharSlice(stackFrame.method), DDOG_PROF_DEMANGLE_OPTIONS_COMPLETE);
 
-                if (result == 0)
+                if (demangleResult.tag == DDOG_PROF_STRING_WRAPPER_RESULT_OK)
                 {
-                    stackFrame.method = std::string(methodData.symbolName);
-
-                    auto demangleResult = ddog_demangle(libdatadog::FfiHelper::StringToCharSlice(stackFrame.method), DDOG_PROF_DEMANGLE_OPTIONS_COMPLETE);
+                    // TODO: There is currently no safe way to free the StringWrapper
+                    auto stringWrapper = demangleResult.ok;
 
-                    if (demangleResult.tag == DDOG_PROF_STRING_WRAPPER_RESULT_OK)
+                    if (stringWrapper.message.len > 0)
                     {
-                        // TODO: There is currently no safe way to free the StringWrapper
-                        auto stringWrapper = demangleResult.ok;
-
-                        if (stringWrapper.message.len > 0)
-                        {
-                            stackFrame.method = std::string((char*)stringWrapper.message.ptr, stringWrapper.message.len);
-                        }
+                        stackFrame.method = std::string((char*)stringWrapper.message.ptr, stringWrapper.message.len);
                     }
                 }
-                else
-                {
-                    std::ostringstream unknownModule;
-                    unknownModule << module.first << "!<unknown>+" << std::hex << (ip - module.second);
-                    stackFrame.method = unknownModule.str();
-                }
             }
-
-            // TODO: Check if the stacktrace is from the tracer or the profiler
-            stackFrame.isSuspicious = false;
-        }
-        else if (resolved == 0)
-        {
-            stackFrame.method = std::string(methodData.symbolName);
-            stackFrame.moduleAddress = methodData.moduleAddress;
-            stackFrame.symbolAddress = methodData.symbolAddress;
-            stackFrame.isSuspicious = methodData.isSuspicious;
+            else
+            {
+                std::ostringstream unknownModule;
+                unknownModule << module.first << "!<unknown>+" << std::hex << (ip - module.second);
+                stackFrame.method = unknownModule.str();
+            }
         }
 
+        // TODO: Check if the stacktrace is from the tracer or the profiler
+        stackFrame.isSuspicious = false;
+
         frames.push_back(stackFrame);
 
     } while (unw_step(&cursor) > 0);
 
-    _UPT_destroy(context);
+    _UPT_destroy(libunwindContext);
+
+    return MergeFrames(frames, managedFrames);
+}
+
+std::vector<StackFrame> CrashReportingLinux::MergeFrames(const std::vector<StackFrame>& nativeFrames, const std::vector<StackFrame>& managedFrames)
+{
+    std::vector<StackFrame> result;
```

Comment:
you can call `reserve(std::max(nativeFrames.size(), managedFrames.size())` to preallocate and avoid temporary allocations.

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/5451/commits/4c07518a1f318dfb013ec2f84173fdc6b5450505